module-generation/scrape.py

```diff
@@ -0,0 +1,67 @@
+import os
+import json
+import requests
+from github import Github
+
+indexDir = "./scrape-data/"
+os.mkdir(indexDir)
+
+accessToken = os.environ.get('GIT_TOKEN')
+g = Github(accessToken)
+
+user = g.get_user('Terasology')
+repo = user.get_repo("Index")
+allModules = repo.get_contents("/distros/omega/gradle.properties")
+getModules = allModules.decoded_content.decode()
+for line in getModules.split():
+    if  line.startswith('extraModules='):
+        filterData=line[13:]
+        modules=filterData.split(",")
+
+# Collect modules 
+for module in range(len(modules)):
+    if(module<len(modules)):
+        repository=user.get_repo(modules[module])
+
+        #Fetch module information
+        try:
+            moduleContent=repository.get_contents("module.txt")
+            getModuleData=moduleContent.decoded_content.decode()
+            parseModuleData=json.loads(getModuleData)
+            moduleName=parseModuleData['id']
+            moduleDir=os.mkdir(indexDir+moduleName)
+            moduleDirSrc=indexDir+moduleName
+            moduleFile= open(moduleDirSrc+"/module.txt", "a")
+            moduleFile.write(getModuleData)
+            moduleFile.close()
+        except:
+            print("Repository is not a Module "+moduleName)
+
+        #Fetch readme file
+        try:
+            readmeContent=repository.get_contents("README.md")
+            getReadmeData=readmeContent.decoded_content.decode()
+            moduleReadmeFile= open(moduleDirSrc+"/README.md", "a")
+            moduleReadmeFile.write(getReadmeData)
+            moduleReadmeFile.close()
+        except:
+            print("No Readme Found "+moduleName)
```

Comment:
the readme not being there is not the only error that can occur and lead into the `except` case... I'd propose to go for "Couldn't fetch README.md" here and also print the error.

Reply:
done